src/js/main.js

```diff
@@ -168,94 +170,145 @@ Site.Map = {
+    _this.mouse.x = event.clientX;
+    _this.mouse.y = event.clientY;
 
-      if (posX >= _this.panZones.left.min && posX <= _this.panZones.left.max) { //left
-        _this.pan.left = true;
-      }
+    if (!_this.isInsidePanZone()) { // If outside pan zones
+      _this.stopPanning(); // Stop animation
+    } else if(!_this.panning) { // If inside and not panning already
+      _this.triggerPanning(); // Trigger animation
+    }
 
-      if (posX >= _this.panZones.right.min && posX <= _this.panZones.right.max) { //right
-        _this.pan.right = true;
-      }
+  },
 
-      _this.triggerAnimation();
-    }
+  stopPanning: function() {
+    var _this =  this;
 
+    _this.panning = false;
   },
 
-  noPan: function() {
+  // (Bool) check if mouse is inside any pan zone
+  isInsidePanZone: function() {
     var _this =  this;
 
-    _this.pan =  {
-      up: false,
-      down: false,
-      left: false,
-      right: false,
-    };
+    var posX = _this.mouse.x;
+    var posY = _this.mouse.y;
+
+    if ( (posX >= _this.panZones.left.min && posX <= _this.panZones.left.max) || // Left
+      (posX >= _this.panZones.right.min && posX <= _this.panZones.right.max) || // Right
+      (posY >= _this.panZones.up.min && posY <= _this.panZones.up.max) || // Up
+      (posY >= _this.panZones.down.min && posY <= _this.panZones.down.max) ) { //  Down
+
+        return true;
+      }
 
+    return false;
   },
 
-  triggerAnimation: function() {
+  // Trigger panning animation
+  triggerPanning: function() {
     var _this =  this;
 
-    window.requestAnimationFrame(_this.animate.bind(_this));
+    _this.panning = true;
+
+    window.requestAnimationFrame(_this.pan.bind(_this));
   },
 
-  animate: function() {
+  // Main panning logic
+  pan: function() {
     var _this =  this;
 
+    // Check is panning has been stopped; we do this here to fail faster
     if (!_this.panning) {
-      return true;
+      return false;
     }
 
-    // Get current map position
+    // Get the angle between center and mouse position
+    var angle = _this.angleFromCenter();
+
+    // Get distance between center and mouse position
+    var distance = _this.distanceFromCenter() * _this.panSpeed;
+
+    // If we don't know the map's current postion we get it
     if (!_this.mapPosition) {
       _this.mapPosition = _this.getMapPosition();
     }
 
-    // Move up
-    if (_this.pan.up && _this.mapPosition[5] <= 0) {
-      _this.mapPosition[5] = _this.mapPosition[5] + 1;
+    // Get new coordinates based on the angle and the distance
+    var newX = _this.mapPosition[4] + (distance * Math.cos(angle));
+    var newY = _this.mapPosition[5] + (distance * Math.sin(angle));
+
+    // Check for left limit
+    if (newX >= 0) {
+      newX = 0;
     }
 
-    // Move down
-    if (_this.pan.down && _this.mapPosition[5] >= (_this.window.height * -2)) {
-      _this.mapPosition[5] = _this.mapPosition[5] - 1;
+    // Check for right limit
+    if (newX <= _this.window.width * -2) {
```

Comment:
That's how transform works. The transform only happens in negative values in this case. `transform: translate(0,0)` means the default position, top-left aligned. Our range of movement goes from 0 to -2windowWitdth for x-axis and from 0 to -2windowHeight.